packages/grid/x-data-grid/src/hooks/features/columnMenu/columnMenuInterfaces.ts

```diff
@@ -0,0 +1,28 @@
+import { OverridableStringUnion } from '@mui/types';
+import React from 'react';
+
+export interface GridColumnMenuState {
+  open: boolean;
+  field?: string;
+}
+
+// To be used by Pro and Premium packages for overriding
+export interface GridColumnMenuKeysInternalOverrides {}
+
+// To be used by users for overriding
+export interface GridColumnMenuKeysOverrides {}
+
+export interface GridColumnMenuTypes {
+  key: OverridableStringUnion<
+    OverridableStringUnion<
+      'filter' | 'sorting' | 'hideColumn' | 'manageColumns' | 'divider',
+      GridColumnMenuKeysInternalOverrides
+    >,
+    GridColumnMenuKeysOverrides
+  >;
+}
+
+export interface GridColumnMenuValue {
+  items: { [key in GridColumnMenuTypes['key']]: React.ReactNode };
+  visibleItemKeys: Array<GridColumnMenuTypes['key']>;
+}
```

Comment:
Since there's only one field, we can get rid of the interface wrapper altogether:

```suggestion
export type GridColumnMenuKeys = OverridableStringUnion<
  OverridableStringUnion<
    'filter' | 'sorting' | 'hideColumn' | 'manageColumns' | 'divider',
    GridColumnMenuKeysInternalOverrides
  >,
  GridColumnMenuKeysOverrides
>;

export interface GridColumnMenuValue {
  items: { [key in GridColumnMenuKeys]: React.ReactNode };
  visibleItemKeys: Array<GridColumnMenuKeys>;
}
```

Reply:
Yep nice catch, thanks, previously I had `type` here, but I was trying to figure out the `OverrideableProps` issue by trying interfaces